<launch>

	<!--	###################	PITT PARAMETERS	#########################	(do not put any " " on values) -->

	<!-- 			ARGS TO THAT CANNOT BE CHANGED AFTER LAUNCH (comment (or ".") (or <0.0f) (or ) for defaults values)	-->

    <param name="/pitt/ref_frame/input_cloud" type="str" value= "/cameraB_depth_optical_frame" />
    <param name="/pitt/ref_frame/output_cloud" type="str" value= "." />
    <param name="/pitt/service/deep_filter/z_threshold" type="double" value= "1.5" />

	<?ignore
	    <!-- List of parameters. Copy-paste the values you want to tweak outside of this block. Default values are reported for reference. -->

        <param name="/pitt/ref_frame/input_cloud" type="str" value= "/cameraB_depth_optical_frame" /> 		<!-- default: "/camera_depth_optical_frame" (used by obj_segmentation node)	-->
        <param name="/pitt/ref_frame/output_cloud" type="str" value= "." />							 		<!-- default: "/world" (used by obj_segmentation node)						-->
        <param name="/pitt/service/deep_filter/z_threshold" type="double" value= "1.5" />					<!-- default: 3.0 (used by obj_segmentation node to call deep filter server)-->

        <rosparam param="/pitt/srv/arm_filter/min_forearm_box">[]</rosparam>							<!-- default:[-0.040, -0.120, -0.190] -->
        <rosparam param="/pitt/srv/arm_filter/max_forearm_box">[]</rosparam>							<!-- default:[0.340,  0.120,  0.105]  -->
       	<rosparam param="/pitt/srv/arm_filter/min_elbow_box">[]</rosparam>								<!-- default:[-0.090, -0.135, -0.160] -->
       	<rosparam param="/pitt/srv/arm_filter/max_elbow_box">[]</rosparam>								<!-- default:[ 0.440,  0.135,  0.110] -->

       	<rosparam param="/pitt/srv/cluster_segmentation/tolerance">[]</rosparam>                        <!-- default:[0.03] -->
       	<rosparam param="/pitt/srv/cluster_segmentation/min_rate">[]</rosparam>                         <!-- default:[0.01] -->
        <rosparam param="/pitt/srv/cluster_segmentation/max_rate">[]</rosparam>                         <!-- default:[0.99] -->
        <rosparam param="/pitt/srv/cluster_segmentation/min_input_size">[]</rosparam>                   <!-- default:[30] -->

        <rosparam param="/pitt/srv/sphere_segmentation/normal_distance_weight">[]</rosparam>             <!-- default:[0.001] -->
        <rosparam param="/pitt/srv/sphere_segmentation/distance_th">[]</rosparam>                       <!-- default:[0.007] -->
       	<rosparam param="/pitt/srv/sphere_segmentation/max_iter_limit">[]</rosparam>                    <!--  default:[1000] -->
       	<rosparam param="/pitt/srv/sphere_segmentation/min_radius_limit">[]</rosparam>                  <!-- default:[0.005] -->
       	<rosparam param="/pitt/srv/sphere_segmentation/max_radius_limit">[]</rosparam>                  <!-- default:[0.500] -->
       	<rosparam param="/pitt/srv/sphere_segmentation/eps_angle_th">[]</rosparam>                      <!--  default:[0.0]  -->
       	<rosparam param="/pitt/srv/sphere_segmentation/min_opening_angle_deg">[]</rosparam>             <!-- default:[100.0] -->
       	<rosparam param="/pitt/srv/sphere_segmentation/max_opening_angle_deg">[]</rosparam>             <!-- default:[180.0] -->
       	<rosparam param="/pitt/srv/sphere_segmentation/min_inliers">[]</rosparam>                       <!--  default:[40]   -->

       	<rosparam param="/pitt/srv/cylinder_segmentation/normal_distance_weight">[]</rosparam>          <!-- default:[0.001] -->
        <rosparam param="/pitt/srv/cylinder_segmentation/distance_th">[]</rosparam>                     <!-- default:[0.008] -->
        <rosparam param="/pitt/srv/cylinder_segmentation/max_iter_limit">[]</rosparam>                  <!--  default:[1000] -->
        <rosparam param="/pitt/srv/cylinder_segmentation/min_radius_limit">[]</rosparam>                <!-- default:[0.005] -->
        <rosparam param="/pitt/srv/cylinder_segmentation/max_radius_limit">[]</rosparam>                <!-- default:[0.500] -->
        <rosparam param="/pitt/srv/cylinder_segmentation/eps_angle_th">[]</rosparam>                    <!-- default:[0.0001] -->
        <rosparam param="/pitt/srv/cylinder_segmentation/min_opening_angle_deg">[]</rosparam>           <!-- default:[50.0]  -->
        <rosparam param="/pitt/srv/cylinder_segmentation/max_opening_angle_deg">[]</rosparam>           <!-- default:[180.0] -->
        <rosparam param="/pitt/srv/cylinder_segmentation/min_inliers">[]</rosparam>                     <!--  default:[40]  -->

        <rosparam param="/pitt/srv/cone_segmentation/normal_distance_weight">[]</rosparam>              <!-- default:[0.0006] -->
        <rosparam param="/pitt/srv/cone_segmentation/distance_th">[]</rosparam>                         <!-- default:[0.0055] -->
        <rosparam param="/pitt/srv/cone_segmentation/max_iter_limit">[]</rosparam>                      <!-- default:[1000] -->
        <rosparam param="/pitt/srv/cone_segmentation/min_radius_limit">[]</rosparam>                    <!-- default:[0.001] -->
        <rosparam param="/pitt/srv/cone_segmentation/max_radius_limit">[]</rosparam>                    <!-- default:[0.500] -->
        <rosparam param="/pitt/srv/cone_segmentation/eps_angle_th">[]</rosparam>                        <!-- default:[0.4] -->
        <rosparam param="/pitt/srv/cone_segmentation/min_opening_angle_deg">[]</rosparam>               <!-- default:[10.0] -->
        <rosparam param="/pitt/srv/cone_segmentation/max_opening_angle_deg">[]</rosparam>               <!-- default:[170.0] -->
        <rosparam param="/pitt/srv/cone_segmentation/min_inliers">[]</rosparam>                         <!-- default:[40] -->

        <rosparam param="/pitt/srv/plane_segmentation/normal_distance_weight">[]</rosparam>             <!-- default:[0.001] -->
        <rosparam param="/pitt/srv/plane_segmentation/distance_th">[]</rosparam>                        <!-- default:[0.007] -->
        <rosparam param="/pitt/srv/plane_segmentation/max_iter_limit">[]</rosparam>                     <!-- default:[1000] -->
        <rosparam param="/pitt/srv/plane_segmentation/eps_angle_th">[]</rosparam>                       <!-- default:[0.0] -->
        <rosparam param="/pitt/srv/plane_segmentation/min_opening_angle_deg">[]</rosparam>              <!-- default:[0.0] -->
        <rosparam param="/pitt/srv/plane_segmentation/max_opening_angle_deg">[]</rosparam>              <!-- default:[10.0] -->
        <rosparam param="/pitt/srv/plane_segmentation/min_inliers">[]</rosparam>                        <!-- default:[40] -->

        <rosparam param="/pitt/srv/geometric_tracker/tracker_forget_threshold">[]</rosparam>            <!-- default:[3] -->
        <rosparam param="/pitt/srv/geometric_tracker/range_threshold">[]</rosparam>                     <!-- default:[0.2] -->
        <rosparam param="/pitt/srv/geometric_tracker/confidence_weight_old">[]</rosparam>               <!-- default:[0.6] -->
        <rosparam param="/pitt/srv/geometric_tracker/confidence_weight_new">[]</rosparam>              <!-- default:[0.4] -->


        <rosparam param="/pitt/srv/supports_segmentation/min_iter_cloud_percent">[]</rosparam>
        <rosparam param="/pitt/srv/supports_segmentation/min_iter_support_percent">[]</rosparam>
        <rosparam param="/pitt/srv/supports_segmentation/horizontal_variance_th">[]</rosparam>
        <rosparam param="/pitt/srv/supports_segmentation/in_shape_distance_th">[]</rosparam>
        <rosparam param="/pitt/srv/supports_segmentation/normal_distance_weight">[]</rosparam>
        <rosparam param="/pitt/srv/supports_segmentation/max_iter">[]</rosparam>
        <rosparam param="/pitt/srv/supports_segmentation/horizontal_axis">[]</rosparam>
        <rosparam param="/pitt/srv/supports_segmentation/edge_remove_offset">[]</rosparam>
    ?>


	<!--			ARGS FOR OBJ_SEGMENTATION NODE	(considered only on launching)	("." for default)				-->
	<arg name="pitt_input_cloud_topic" value= "/cameraB/depth/points" /> 			<!-- default: "/camera/depth/points"					-->
	<arg name="pitt_show_original_cloud" value="1" /> 								<!-- default: "0" (false) [white ptn]					-->
	<arg name="pitt_show_supports" value="." /> 									<!-- default: "0" (false) [brown ptn]					-->
	<arg name="pitt_show_cluster" value="." />		 								<!-- default: "0" (false) [random ptn]					-->
	<arg name="pitt_show_object_on_support" value="." /> 							<!-- default: "0" (false) [orange ptn]					-->
	<arg name="pitt_raw_centroid_log_file_path" value="." /> 						<!-- default: "" (do not print)							-->
		  					<!-- set to: ".." for timed formatting name (works only at the end of the path) or set to any other paths.
		  						 ExAMPLE <arg name="pitt_raw_centroid_log_file_path" value="$(find pitt_object_table_segmentation)/logs/log-.." />
		  					-->

	<arg name="pitt_arm_filter_show_cloud" value= "." /> 										<!-- default:"0"(false)[red arm][green objs]-->
	<arg name="pitt_arm_filter_camera_frame" value= "/cameraB_depth_optical_frame" /> 			<!-- default: "/camera_depth_optical_frame"	-->
	<arg name="pitt_arm_filter_left_forearm_frame" value= "." /> 								<!-- default: "/left_lower_forearm"			-->
	<arg name="pitt_arm_filter_right_forearm_frame" value= "." /> 								<!-- default: "/right_lower_forearm"		-->
	<arg name="pitt_arm_filter_left_elbow_frame" value= "." /> 									<!-- default: "/left_lower_elbow"			-->
	<arg name="pitt_arm_filter_right_elbow_frame" value= "." /> 								<!-- default: "/right_lower_elbow"			-->

	<!--			ARGS FOR RANSAC_SEGMENTATION NODE	(considered only on launching)	("." for default)				-->

	<arg name="ransac_show_primitives" value= "." /> 										<!-- default:"0"(false)-->

	<!--			ARGS FOR GEOMETRIC_TRACKER NODE	(considered only on launching)	("." for default)				-->

	<arg name="show_geometric_tracker" value= "." /> 										<!-- default:"0"(false)-->


		<!--	###################	PITT LAUNCH #########################	-->

		<!-- 		    KINECT DRIVER LAUNCH	 																		-->
	<!-- <include file="$(find freenect_launch)/launch/freenect.launch"/> -->



		<!-- 			CLUSTER AND SUPPORT CONFIG																		-->
	<node name="deep_filter_srv" 				pkg="pitt_object_table_segmentation" 	type="deep_filter_srv" 				/> <!-- output="screen"/> -->
	<node name="arm_filter_srv" 				pkg="pitt_object_table_segmentation" 	type="arm_filter_srv"
			args=	"$(arg pitt_arm_filter_show_cloud)
					 $(arg pitt_arm_filter_camera_frame)
					 $(arg pitt_arm_filter_left_forearm_frame)
					 $(arg pitt_arm_filter_right_forearm_frame)
					 $(arg pitt_arm_filter_left_elbow_frame)
					 $(arg pitt_arm_filter_right_elbow_frame)"
		/> <!-- output="screen"/> -->
	<node name="supports_segmentation_srv" 		pkg="pitt_object_table_segmentation" 	type="supports_segmentation_srv" 	/> <!-- output="screen"/> -->
	<node name="cluster_segmentation_srv" 		pkg="pitt_object_table_segmentation" 	type="cluster_segmentation_srv"		/> <!-- output="screen"/> -->
	<node name="obj_segmentation" 				pkg="pitt_object_table_segmentation" 	type="obj_segmentation"
			output="screen"
			args=	"$(arg pitt_input_cloud_topic)
					 $(arg pitt_show_original_cloud)
					 $(arg pitt_show_supports)
					 $(arg pitt_show_cluster)
					 $(arg pitt_show_object_on_support)
					 $(arg pitt_raw_centroid_log_file_path)"
		/>



		<!-- 			RANSAC CONFIG																					-->
	<node name="plane_segmentation_srv" 		pkg="pitt_object_table_segmentation" 	type="plane_segmentation_srv"		/> <!-- output="screen"/> -->
	<node name="sphere_segmentation_srv" 		pkg="pitt_object_table_segmentation" 	type="sphere_segmentation_srv" 		/> <!-- output="screen"/> -->
	<node name="cylinder_segmentation_srv" 		pkg="pitt_object_table_segmentation" 	type="cylinder_segmentation_srv" 	/> <!-- output="screen"/> -->
	<node name="cone_segmentation_srv" 			pkg="pitt_object_table_segmentation" 	type="cone_segmentation_srv" 		/> <!-- output="screen"/> -->
	<node name="ransac_segmentation" 			pkg="pitt_object_table_segmentation" 	type="ransac_segmentation" 			output="screen"	    args= "$(arg ransac_show_primitives)"/>



		<!--			TRACKER																							-->
	<node name="geometric_tracker" 				pkg="pitt_geometric_tracking" 			type="geometric_tracker"
		args= "$(arg show_geometric_tracker)"		/> <!-- output="screen"/> -->

</launch>
